### PR TITLE
TSD-163 fix legend

### PR DESCRIPTION
**Pull request by esoliani**

### CODE REVIEW

File: app/scripts/controllers/dashboard.coffee [hunk 1]
```diff
@@ -88,10 +88,15 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
         watchCompute = watchComputeSrv $scope
 
         watchCompute 'metric',
-          dependencies: ['schoolType', 'rankBy']
-          computer: ([schoolType, criteria]) ->
+          dependencies: ['schoolType', 'rankBy', 'visMode']
+          computer: ([schoolType, criteria, visMode], [oldSchoolType, oldCriteria, oldVisMode]) ->
             unless schoolType? and criteria?
               null
+            else if oldVisMode? and  visMode != oldVisMode
```
Comment (esoliani) [inline app/scripts/controllers/dashboard.coffee]: should this be done on the brackets service?

Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee]: Hmm, yeah maybe. The logic is so complex you know :stuck_out_tongue: 

the `visMode != oldVisMode` seems suspicious to me. Maybe we can just change `brackets.getMetric` to (optionally?) take `visMode` as a param?

File: app/views/dashboard.html [hunk 2]
```diff
@@ -10,6 +10,7 @@
       <p class="performance-select">
         <a ng-class="{active: visMode==='passrate'}" ng-click="setVisMode('passrate')" translate="legend.passrate">Pass Rate (%)</a>
         <a ng-class="{active: visMode==='ptratio'}" ng-click="setVisMode('ptratio')" translate="legend.ptratio">Pupil-Teacher Ratio</a>
+        <a ng-hide="schoolType==='primary'" ng-class="{active: visMode==='gpa'}" ng-click="setVisMode('gpa')" translate="legend.gpa">GPA</a>
```
Comment (uniphil) [inline app/views/dashboard.html]: I think `ng-show` for `secondary` schoolType is a little more direct instead of `ng-hide`.

Comment (esoliani) [inline app/views/dashboard.html]: fixed

File: app/scripts/controllers/dashboard.coffee (hunk 1 above)
Comment (esoliani) [inline app/scripts/controllers/dashboard.coffee]: refactoring as talk

**Pull request closed without merging**